### PR TITLE
Generations Command "advance-generation" Becomes "track"

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

Depends on #9972.

This patch renames the `advance-generation` command to `track`, including:
- File-name changes.
- Help text and command description changes.
- API method renaming.
- Modified comments.
- Validation of unit existence and life when adding a unit to the generation document.

## QA steps

- `export JUJU_DEV_FEATURE_FLAGS=generations`
- Bootstrap.
- `juju branch test-branch`
- `juju deploy redis`
- `juju track test-branch riak` results in `ERROR application "riak" not found`.
- `juju track test-branch redis/1` results in `ERROR unit "redis/1" not found`.
- `juju track test-branch redis/0`
- `juju show-generation test-branch` (command rename pending):
```
test-branch:
  created: 28 Mar 2019 13:41:43+01:00
  created-by: admin
  applications:
  - application: redis
    units:
    - redis/0
    config: {}
```

## Documentation changes

None

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Command "advance-generation" -> "track"](https://trello.com/c/YraYku46/1232-command-advance-generation-track)

File: cmd/juju/model/trackbranch.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	trackBranchSummary = "Set units and/or applications to realise changes made under a branch."
+	trackBranchDoc     = `
+Specific units can be set to track a branch by supplying multiple unit IDs.
+All units of an application can be set to track a branch by passing an
+application name. Units can only track one branch at a time.
+
+Examples:
+    juju track test-branch redis/0
+    juju track test-branch redis
+    juju track test-branch redis -n 2
```
Comment (manadart) [inline cmd/juju/model/trackbranch.go]: Note that this mode is not yet implemented, but forthcoming.

File: cmd/juju/model/trackbranch.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    diff
+`
+)
+
+// NewTrackBranchCommand wraps trackBranchCommand with sane model settings.
+func NewTrackBranchCommand() cmd.Command {
+	return modelcmd.Wrap(&trackBranchCommand{})
+}
+
+// trackBranchCommand supplies the "track" CLI command used to make units \
```
Comment (SimonRichardson) [inline cmd/juju/model/trackbranch.go]: nit `\` <- not required

Comment (manadart) [reply cmd/juju/model/trackbranch.go]: Vim search in edit mode :)

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
```
2019-04-03 16:29:04 WARNING juju.testing mgo.go:212 failed to start mongo: mongod failed to listen on port 41451
```
$$merge$$

**Comment by manadart** [comment]
Hitting all the know intermittents.
$$merge$$

**Pull request merged**